cpp/installer.cpp

```diff
@@ -293,106 +337,110 @@ void installSequel(jsi::Runtime &rt, const char *docPath)
+                                 {
+              if(result.type == SequelResultOk)
               {
-                SequelLiteralUpdateResult result = sequel_execute_literal_update(dbName, line);
-                if (result.type == SequelResultError)
-                {
-                  sequel_execute_literal_update(dbName, "ROLLBACK");
-                  auto res = jsi::Object(rt);
-                  res.setProperty(rt, "status", jsi::Value(1));
-                  res.setProperty(rt, "message", jsi::String::createFromUtf8(rt, result.message.c_str()));
-                  sqFile.close();
-                  return move(res);
-                }
-                else
-                {
-                  affectedRows += result.affectedRows;
-                  commands++;
-                }
-              }
-            }
-            sqFile.close();
-            sequel_execute_literal_update(dbName, "COMMIT");
-            auto res = jsi::Object(rt);
-            res.setProperty(rt, "status", jsi::Value(0));
-            res.setProperty(rt, "rowsAffected", jsi::Value(affectedRows));
-            res.setProperty(rt, "commands", jsi::Value(commands));
-            return move(res);
+                auto res = jsi::Object(rt);
+                res.setProperty(rt, "status", jsi::Value(0));
+                res.setProperty(rt, "rowsAffected", jsi::Value(result.affectedRows));
+                res.setProperty(rt, "commands", jsi::Value(result.commands));
+                callback->asObject(rt).asFunction(rt).call(rt, move(res));
+              } else {
+                callback->asObject(rt).asFunction(rt).call(rt, createError(rt, result.message));
+              } });
           }
-          catch (...)
+          catch (std::exception &exc)
           {
-            sqFile.close();
-            sequel_execute_literal_update(dbName, "ROLLBACK");
-            return createError(rt, "[react-native-quick-sqlite][loadSQLFile] Unexpected error, transaction was rolledback");
+            LOGW("Catched exception: %s", exc.what());
+            invoker->invokeAsync([&rt, err = exc.what(), callback]
+                                 { callback->asObject(rt).asFunction(rt).call(rt, createError(rt, "Unknow error")); });
           }
-        }
-        else
-        {
-          return createError(rt, "[react-native-quick-sqlite][loadSQLFile] Could not open file");
-        }
+        };
+        pool->queueWork(task);
+        return {};
       });
 
   // Async Execute SQL
   auto asyncExecSQL = jsi::Function::createFromHostFunction(
       rt,
       jsi::PropNameID::forAscii(rt, "sequel_asyncExecSQL"),
-      3,
-      [](jsi::Runtime &rt, const jsi::Value &thisValue, const jsi::Value *args, size_t count) -> jsi::Value
+      4,
+      [pool](jsi::Runtime &rt, const jsi::Value &thisValue, const jsi::Value *args, size_t count) -> jsi::Value
       {
+        if (count < 4)
+        {
+          LOGW("NOT ENOUGH PARAMS PASSED");
```

Comment:
Add proper validation

Reply:
Added one error throw, because if no callback is passed, the return value with the `status` field cannot be returned to the user. I think it's ok to throw an error on this case.